python/ray/experimental/ui.py

```diff
@@ -0,0 +1,722 @@
+                                                           + diff * high))
+
+                # (Querying based on % of total number of tasks that were
+                # run.)
+                elif breakdown_opt.value == total_tasks_value:
+                    if range_slider.value[0] == 0:
+                        tasks = ray.global_state.task_profiles(num_tasks=(int(
+                                                               num_tasks
+                                                               * high)),
+                                                               fwd=True)
+                    else:
+                        tasks = ray.global_state.task_profiles(num_tasks=(int(
+                                                               num_tasks *
+                                                               (high - low))),
+                                                               fwd=False)
+
+                update(smallest, largest, num_tasks, tasks)
+
+    # Get updated values from a slider or text box, and update the rest of
+    # them accordingly.
+    range_slider.observe(update_wrapper, names="value")
+    breakdown_opt.observe(update_wrapper, names="value")
+    start_box.observe(update_wrapper, names="value")
+    end_box.observe(update_wrapper, names="value")
+    num_tasks_box.observe(update_wrapper, names="value")
+
+    # Initializes the sliders
+    update_wrapper(INIT_EVENT)
+
+    # Display sliders and search boxes
+    display(start_box, end_box, range_slider, num_tasks_box, breakdown_opt)
+
+    # Return the sliders and text boxes
+    return start_box, end_box, range_slider, breakdown_opt
+
+
+def object_search_bar():
+    object_search = widgets.Text(
+        value="",
+        placeholder="Object ID",
+        description="Search for an object:",
+        disabled=False
+        )
+    display(object_search)
+
+    def handle_submit(sender):
+        pp = pprint.PrettyPrinter()
+        pp.pprint(ray.global_state.object_table(object_search.value))
+
+    object_search.on_submit(handle_submit)
+
+
+def task_search_bar():
+    task_search = widgets.Text(
+        value="",
+        placeholder="Task ID",
+        description="Search for a task:",
+        disabled=False
+        )
+    display(task_search)
+
+    def handle_submit(sender):
+        pp = pprint.PrettyPrinter()
+        pp.pprint(ray.global_state.task_table(task_search.value))
+
+    task_search.on_submit(handle_submit)
+
+
+def task_timeline():
+    path_input = widgets.Button(description="View task timeline")
+
+    breakdown_basic = "Basic"
+    breakdown_task = "Task Breakdowns"
+
+    breakdown_opt = widgets.Dropdown(
+        options=["Basic", "Task Breakdowns"],
+        value="Basic",
+        description="View options:",
+        disabled=False,
+    )
+
+    start_box, end_box, range_slider, time_opt = get_sliders(False)
+    # display(breakdown_opt)
+    # display(path_input)
```

Comment:
@alanamarzoev Is this a bug. Should `display(path_input)` be commented in? The button for viewing the timeline seems to have disappeared.